In [1]:
import pandas as pd
import numpy as np

In [13]:
titan_df = pd.read_csv('./train.csv', engine='python')
print(titan_df)

     PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ... 

In [14]:
pd.read_csv('./train.csv', engine='python')

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [4]:
### 60세 이상의 데이터만 가져와주세요 ^^
### 풀이 -1
len(titan_df[titan_df['Age'] >= 60])

26

In [5]:
### 60세 이상의 데이터만 가져와 주세요 ^^
### 풀이 -2
titan_df[titan_df['Age']>=60].shape[0]

26

In [7]:
### 60세 이상의 데이터 중에서 Name, Age, Survived만 가져와주세요.
### 풀이 -1
new_df = titan_df[titan_df['Age']>=60]
new_df[['Name', 'Age', 'Survived']]

Name   Age  Survived
33                              Wheadon, Mr. Edward H  66.0         0
54                     Ostby, Mr. Engelhart Cornelius  65.0         0
96                          Goldschmidt, Mr. George B  71.0         0
116                              Connors, Mr. Patrick  70.5         0
170                         Van der hoef, Mr. Wyckoff  61.0         0
252                         Stead, Mr. William Thomas  62.0         0
275                 Andrews, Miss. Kornelia Theodosia  63.0         1
280                                  Duane, Mr. Frank  65.0         0
326                         Nysveen, Mr. Johan Hansen  61.0         0
366  Warren, Mrs. Frank Manley (Anna Sophia Atkinson)  60.0         1
438                                 Fortune, Mr. Mark  64.0         0
456                         Millet, Mr. Francis Davis  65.0         0
483                            Turkula, Mrs. (Hedwig)  63.0         1
493                           Artagaveytia, Mr. Ramon  71.0         0
545                      Nicholson, Mr. Arthur Ernest  64.0         0
555                                Wright, Mr. George  62.0         0
570                                Harris, Mr. George  62.0         1
587                  Frolicher-Stehli, Mr. Maxmillian  60.0         1
625                             Sutton, Mr. Frederick  61.0         0
630              Barkworth, Mr. Algernon Henry Wilson  80.0         1
672                       Mitchell, Mr. Henry Michael  70.0         0
684                 Brown, Mr. Thomas William Solomon  60.0         0
694                                   Weir, Col. John  60.0         0
745                      Crosby, Capt. Edward Gifford  70.0         0
829         Stone, Mrs. George Nelson (Martha Evelyn)  62.0         1
851                               Svensson, Mr. Johan  74.0         0

In [10]:
### 60세 이상의 데이터 중에서 Name, Age, Survived만 가져와주세요.
### 풀이 -2
titan_df[titan_df['Age']>=60][['Name','Age','Survived']]

Name   Age  Survived
33                              Wheadon, Mr. Edward H  66.0         0
54                     Ostby, Mr. Engelhart Cornelius  65.0         0
96                          Goldschmidt, Mr. George B  71.0         0
116                              Connors, Mr. Patrick  70.5         0
170                         Van der hoef, Mr. Wyckoff  61.0         0
252                         Stead, Mr. William Thomas  62.0         0
275                 Andrews, Miss. Kornelia Theodosia  63.0         1
280                                  Duane, Mr. Frank  65.0         0
326                         Nysveen, Mr. Johan Hansen  61.0         0
366  Warren, Mrs. Frank Manley (Anna Sophia Atkinson)  60.0         1
438                                 Fortune, Mr. Mark  64.0         0
456                         Millet, Mr. Francis Davis  65.0         0
483                            Turkula, Mrs. (Hedwig)  63.0         1
493                           Artagaveytia, Mr. Ramon  71.0         0
545                      Nicholson, Mr. Arthur Ernest  64.0         0
555                                Wright, Mr. George  62.0         0
570                                Harris, Mr. George  62.0         1
587                  Frolicher-Stehli, Mr. Maxmillian  60.0         1
625                             Sutton, Mr. Frederick  61.0         0
630              Barkworth, Mr. Algernon Henry Wilson  80.0         1
672                       Mitchell, Mr. Henry Michael  70.0         0
684                 Brown, Mr. Thomas William Solomon  60.0         0
694                                   Weir, Col. John  60.0         0
745                      Crosby, Capt. Edward Gifford  70.0         0
829         Stone, Mrs. George Nelson (Martha Evelyn)  62.0         1
851                               Svensson, Mr. Johan  74.0         0

여러 개의 복합 조건도 결합해 적용할 수 있다.

1) and 조건일 때는 &

2) or 조건일 때는 |

3) Not 조건일 때는 ~

### 나이가 60세 초과이고 선실등급이 1등급이며, 성별이 여성인 승객을 추출해보자. 개별조건은 ()로 묶어서 복합연산자를 활용하여 정답을 낼 수 있다.

In [18]:
# 정답은 2~
con1 = titan_df.Age > 60
con2 = titan_df.Pclass == 1
con3 = titan_df.Sex == 'female'
print(titan_df[con1&con2&con3])

     PassengerId  Survived  Pclass                                       Name  \
275          276         1       1          Andrews, Miss. Kornelia Theodosia   
829          830         1       1  Stone, Mrs. George Nelson (Martha Evelyn)   

        Sex   Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
275  female  63.0      1      0   13502  77.9583    D7        S  
829  female  62.0      0      0  113572  80.0000   B28      NaN  


### 위의 문제를 반대로 하여 총 row가 889명이 출력되도록 풀이해보세요 ^^

In [19]:
titan_df[(titan_df['Age']>60) & (titan_df['Pclass'] == 1) & (titan_df['Sex'] == 'female')]

PassengerId  Survived  Pclass                                       Name  \
275          276         1       1          Andrews, Miss. Kornelia Theodosia   
829          830         1       1  Stone, Mrs. George Nelson (Martha Evelyn)   

        Sex   Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
275  female  63.0      1      0   13502  77.9583    D7        S  
829  female  62.0      0      0  113572  80.0000   B28      NaN

In [40]:
print(titan_df.shape)
titan_df[~((titan_df['Age']>60) & (titan_df['Pclass'] == 1) & (titan_df['Sex'] == 'female'))]
# type(titan_df)
# titan_df.delete(t1)
# del titan_df[t1]
# print(titan_df.shape)
# print(t1)

(891, 12)


PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[889 rows x 12 columns]